language/move-lang/src/typing/translate.rs

```diff
@@ -760,13 +759,13 @@ fn exp_(context: &mut Context, sp!(eloc, ne_): N::Exp) -> T::Exp {
                 );
                 (idx, (fty, arg))
             });
-            if m != current_module {
-                let msg = "Currently, all structs can only be instantiated in module that they \
-                           were declared";
-                context.error(vec![
-                    (eloc, format!("Invalid instantiation of '{}::{}'", &m, &n)),
-                    (current_module.loc(), msg.into()),
-                ])
+            if !context.is_current_module(&m) {
+                let msg = format!(
+                    "Invalid instantiation of '{}::{}'.\n\
+                     All structs can only be deconstructed in module that they were declared",
```

Comment:
This is fine as-is, with the exception that this is an error in construction rather than deconstruction.

What do you think about: "... All structs can only be constructed in the module in which they are declared"?